src/bidiMapper/domains/context/CdpTarget.ts

```diff
@@ -173,6 +174,27 @@ export class CdpTarget {
     });
   }
 
+  async toggleNetworkIfNeeded(): Promise<void> {
+    const enableNetwork =
+      this.#eventManager.subscriptionManager.isSubscribedToModule(
+        this.#id,
+        BiDiModule.Network
+      );
+
+    try {
+      if (enableNetwork && !this.#networkDomainEnabled) {
+        this.#networkDomainEnabled = true;
+        await this.#cdpClient.sendCommand('Network.enable');
```

Comment:
Maybe more atomic?
```suggestion
        await this.#cdpClient.sendCommand('Network.enable').catch(e=>{
            this.#networkDomainEnabled = false;
            throw e;
        |});
```

Reply:
I think the suggestion here is NoOp.

I am not sure about this the current implementation does the the following.
If changes -> enable/disable and update flag.
If no changes -> no error can be thrown so we never go to catch.
If command fails -> flip to previous state.

If second change comes while we are waiting for response we should be good as well as both commands will be send to the same process in Chrome and will be in queue from what i understand.

Reply:
Agree that your solution will work, but I'm afraid it's easier to screw up with it later on. Up for you.